src/core/globals/blockchain.c

```diff
@@ -0,0 +1,68 @@
+#include "blockchain.h"
+
+void blockchain_init() {
+  Block* genesis_block;
+
+  blockchain = NULL;  // Must always start at NULL
+
+  genesis_block = malloc(sizeof(Block));
+
+  // Initialize with a totally empty block
+  genesis_block->num_txs = 0;
+  genesis_block->txs = NULL;
+
+  genesis_block->header.timestamp = 0;
+  genesis_block->header.nonce = 0;
+  memset(&(genesis_block->header.all_tx), 0, TX_HASH_LEN);
+  memset(&(genesis_block->header.prev_header_hash), 0, BLOCK_HASH_LEN);
+
+  blockchain_add(genesis_block);
+}
+
+Block *blockchain_add(Block *block) {
```

Comment:
good naming scheme of ```structure_action()``` I will try to follow this pattern. although we also have things like ser_tx and ser_block....thoughts?

Reply:
I think its weirder because there are more "actions of the object" (ex: blockchain removing a node), versus actions on the object? (ex: serializing a transaction) No idea if that distinction makes sense of what we do about it haha